apps/shipments/views/permission_link.py

```diff
@@ -66,16 +68,31 @@ def create(self, request, *args, **kwargs):
 
         if settings.PROFILES_ENABLED and emails:
             LOG.debug(f'Emailing permission link: {permission_link.id}')
-            username = request.user.username
-            protocol = 'https' if request.is_secure() else 'http'
-            subject = f'{username} shared a shipment details page with you.'
-            email_context = {
-                'username': username,
-                'link': f'{protocol}://{settings.FRONTEND_DOMAIN}/shipments/{shipment_id}/'
-                        f'?permission_link={permission_link.id}',
-                'request': request
-            }
-
-            send_templated_email('email/shipment_link.html', subject, email_context, emails)
+            try:
+                link = f'{"https" if request.is_secure() else "http"}://{settings.FRONTEND_DOMAIN}/shipments/' \
+                       f'{shipment_id}/?permission_link={permission_link.id}'
+                if settings.IOT_THING_INTEGRATION:
+                    url_client = URLShortener()
+                    params = {
+                        'long_url': link,
+                        'expiration_date': (
+                            permission_link.expiration_date.isoformat() if permission_link.expiration_date else None)
+                    }
+
+                    link = url_client.get_shortened_link(params)
+
+                email_context = {
+                    'username': request.user.username,
+                    'link': link,
+                    'request': request
+                }
+
+                send_templated_email('email/shipment_link.html',
+                                     f'{request.user.username} shared a shipment details page with you.',
+                                     email_context,
+                                     emails)
+            except UrlShortenerError as exc:
+                permission_link.delete()
+                raise exc
```

Comment:
I think this exception handling is too wide especially since you are catching just the url shortener exception, the following should be sufficient:
```python
try:
  link = url_client.get_shortened_link(params)
except UrlShortenerError:
  ....
```
Also I do not understand why you delete the permission link in case of exception? I think you could just send the normal link if for some reason the link creation is unsuccessful and log the exception so that we can trouble shut it issue. At least this way we do not run the risk of having this feature unavailable for long period of time if the issue is on the url shortener service.

Reply:
Hmm, that's a good call. I generally just have things not created, but it isn't really necessary here. And good call, I'll shorten up the try catch.